proxy/http/HttpSessionManager.cc

```diff
@@ -75,17 +75,32 @@ ServerSessionPool::match(HttpServerSession *ss, sockaddr const *addr, INK_MD5 co
          (TS_SERVER_SESSION_SHARING_MATCH_HOST == match_style || ats_ip_addr_port_eq(ss->get_server_ip(), addr));
 }
 
+bool
+ServerSessionPool::match_sni(HttpSM *sm, NetVConnection *netvc)
+{
+  // TS-4468: If the connection matches, make sure the SNI server
+  // name (if present) matches the request hostname
+  int len                 = 0;
+  const char *req_host    = sm->t_state.hdr_info.server_request.host_get(&len);
+  const char *session_sni = netvc->options.sni_servername;
+
+  return ((sm->t_state.scheme != URL_WKSIDX_HTTPS) || !session_sni || !strncmp(session_sni, req_host, len));
```

Comment:
If the scheme is not https, there is no need to do the SNI check.  
Agreed, might as well do a case insensitive compare here.